File: MORM.Java/src/MORM/Java/Classes/Collection.java
package MORM.Java.Classes;

import java.util.ArrayList;

import MORM.Java.Classes.Mapping.Relacao;

@SuppressWarnings("serial")
public class Collection extends ArrayList<CollectionItem>
{
    public Collection(Class<CollectionItem> collectionItemClass)
    {
        CollectionItemClass = collectionItemClass;
    }
    
    public Class<CollectionItem> CollectionItemClass;
    
    public CollectionItem AddItem()
    {
        CollectionItem collectionItem = null;
		try {
			collectionItem = CollectionItemClass.newInstance();
	        this.add(collectionItem);
		} catch (InstantiationException e) {
			e.printStackTrace();
		} catch (IllegalAccessException e) {
			e.printStackTrace();
		}
        return collectionItem;
    }
    
    private Relacao relacao;
    
    public void SetRelacao(Object owner, String campos)
    {
    	this.relacao = new Mapping().new Relacao(owner, campos);
    }

    public Relacao GetRelacao()
    {
        return this.relacao;
    }
}

File: MORM.CSharp/MORM.Util/Classes/Collection.cs
using System;
using System.Collections.Generic;

namespace MORM.Util.Classes
{
    public class Collection : List<CollectionItem>
    {
        public Collection(Type collectionItemClass)
        {
            CollectionItemClass = collectionItemClass;
        }
        
        public Type CollectionItemClass { get; private set; }
        
        public CollectionItem AddItem()
        {
            var collectionItem = Activator.CreateInstance(CollectionItemClass) as CollectionItem;
            this.Add(collectionItem);
            return collectionItem;
        }
        
        private Relacao _relacao;
        
        public void SetRelacao(object owner, string campos)
        {
            _relacao = new Relacao(owner, campos);
        }
        
        public Relacao GetRelacao()
        {
            return _relacao;
        }
    }
}